console/console-init/ui/src/modules/device/components/SecretList/SecretList.tsx

```diff
@@ -225,6 +173,19 @@ export const SecretList: React.FC<ISecretListProps> = ({
             </FormGroup>
           </GridItem>
         )}
+        {secrets.length > 1 && (
+          <GridItem span={12}>
+            <Button
+              className={css(styles.delete_secret)}
+              variant="link"
+              type="button"
+              icon={<MinusCircleIcon />}
+              onClick={() => onDeleteSecrets(credentialId, "secrets", id)}
```

Comment:
we should avoid inline functions

Reply:
Here, without inline function we can't get the credentialId value outside of loop. That's why we need it.